Deprecating LC_ALL from locale resource

Nimesh-Msys opened the pull request:
Signed-off-by: Nimesh-Msys <nimesh.patni@msystechnologies.com>

### Description

Added a deprecation doc to inform users that we no longer support changing *lc_all* locale. Instead we may change other specific *lc_* property as needed

### Definition of Done

### Issues Resolved

Based on the [following conversations](https://github.com/chef/chef/pull/8100), it would be a possible solution for [7904](https://github.com/chef/chef/issues/7904)

### Output

![chef_deprecations_client](https://user-images.githubusercontent.com/36878084/54603776-30ab4e80-4a6b-11e9-9421-04e5302c9207.jpg)

![deprecations_lc_all](https://user-images.githubusercontent.com/36878084/54603795-3a34b680-4a6b-11e9-84b2-82d171f46658.jpeg)




### Check List

- [x] Spell Check
- [x] Local build
- [x] Examine the local build
- [x] All tests pass

Nimesh-Msys committed:
Deprecating LC_ALL from locale resource

- Assigned new target ID (27) for lc_all deprecation
- Minor changes as per the review comments.

Signed-off-by: Nimesh-Msys <nimesh.patni@msystechnologies.com>

btm commented on chef_master/source/chef_deprecations_client.rst:
```diff
@@ -163,6 +163,10 @@ All Deprecations
+  * - `CHEF-27 </deprecations_locale_lc_all.html>`__
+    - Deprecation of lc_all from locale resource
+    - 15.0
+    - 15.1
```
Let's plan to deprecate this in 16.0.

btm replied:
for the deprecation version let's set to 14.12, and we will backport the deprecation into the Chef 14 release we will make in April/May. And set the second number to 16.0 as I commented/

btm commented on chef_master/source/deprecations_locale_lc_all.rst:
```diff
@@ -0,0 +1,47 @@
+
+.. end_tag
+
+This deprecation warning was added after Chef 15.0.185. Support for property `lc_all` will likely be removed by Chef 15.1.0.
```
I don't think we need this line, I think this information is generated in the docs from lines in `chef_master/source/chef_deprecations_client.rst`.

btm replied:
Apparently the other file doesn't do anything automated for us, so we need to keep this line. But lets say:

> This deprecation warning was added ln Chef 14.12. Support for property `lc_all` will be removed for Chef 16.0

Nimesh-Msys replied:
Its not automated. Have added the comment manually.

btm commented on chef_master/source/deprecations_locale_lc_all.rst:
```diff
@@ -0,0 +1,47 @@
+
+.. tag deprecations_locale_lc_all
+
+Setting *LC_ALL* variable is NOT recommended and can break Chef's parsing of command output in unexpected ways. Use one of the more specific *LC_* properties as needed.
```
Setting the *LC_ALL* variable is NOT recommended. As a system-wide setting, *LANG* should provide the desired behavior. *LC_ALL* is intended to be used for temporarily troubleshooting issues rather than an every day system setting. Changing *LC_ALL* can break Chef's parsing of command output in unexpected ways. Use one of the more specific *LC_* properties as needed.

btm commented on chef_master/source/deprecations_locale_lc_all.rst:
```diff
@@ -0,0 +1,47 @@
+
+	export LC_ALL="<locale_name>"
+
+1. /etc/default/locale
```
I would add this:

> Setting *LC_* variables varies by platform, but common locations to configure *LC_* variables are:

And then drop the note on "priority order" below.

Nimesh-Msys replied:
Updated

btm commented on chef_master/source/deprecations_locale_lc_all.rst:
```diff
@@ -0,0 +1,47 @@
+	locale -v
+
+.. warning::
+	- Changes on any other file might only be applicable till current session of shell.
```
drop this warning. this is probably a step too far in trying to teach unix than provide a workaround for the deprecation of the `lc_all` property.

Nimesh-Msys replied:
True. Updated

Nimesh-Msys commented:
@btm , I've updated this PR as per the review comments. Tested and updated the output in Description.
Also squashed the changes into single commit. Could you please take a look and let me know in case of any issues.

Commit:
Merge branch 'master' into Nimesh/MSYS-987_deprecate_lc_all

mjingle committed:
Merge branch 'master' into Nimesh/MSYS-987_deprecate_lc_all

mjingle commented:
@Nimesh-Msys Thank you for your documentation contribution and the level of detail in your PR! Because this is a PR from a remote fork, our set-up means that we had to include your work to chef-web-docs through other means. Please see: https://github.com/chef/chef-web-docs/pull/1858
